### PR TITLE
GPU: Document system requirements, add upfront Metal hardware support checks

### PR DESCRIPTION
**Pull request by TheSpydog**
This PR adds documentation for the OS/driver/hardware requirements for each of the SDL_GPU backends. This should hopefully clear up a lot of ambiguity around "is this device supported" and gives developers something concrete to use for their own system requirements and support.

At runtime, to ensure the device meets SDL_GPU driver's system requirements, Vulkan and D3D12 perform a series of feature support queries during the device creation process. However, for Metal, we only performed a single OS version check. This allowed apps to get pretty far into their lifecycle on unsupported devices before they crashed by using an unsupported feature.

To fix this, I've added hardware support queries as early as possible (right after MTLDevice creation) to detect if the app is running on an unsupported device, so that apps can handle the failure gracefully. I also bumped up the minimum macOS version from 10.13 to 10.14 because lower versions don't have the hardware support query we need.

Resolves https://github.com/libsdl-org/SDL/issues/11692

### CODE REVIEW

File: include/SDL3/SDL_gpu.h [hunk 1]
```diff
@@ -159,6 +159,24 @@
  * and reasonable to implement across several platforms and underlying APIs.
  * So while these things are not in the "never" category, they are definitely
  * not "near future" items either.
+ *
+ * ## System Requirements
+ *
+ * **Vulkan:** Vulkan 1.0 with the following extensions and device features:
+ * - `VK_KHR_swapchain`
```
Comment (flibitijibibo) [inline include/SDL3/SDL_gpu.h:166]: I wonder if it's worth fussing with this for windowless rendering, probably better to wait for the wildly unlikely scenario where this doesn't exist?

File: include/SDL3/SDL_gpu.h [hunk 2]
```diff
@@ -159,6 +159,24 @@
...
+ * - `VK_KHR_maintenance1`
+ * - `independentBlend`
+ * - `imageCubeArray`
+ * - `depthClamp`
+ * - `shaderClipDistance`
+ * - `drawIndirectFirstInstance`
+ *
+ * **D3D12:** Windows 10 or newer, with DirectX 12 Feature Level 11_1
```
Comment (flibitijibibo) [inline include/SDL3/SDL_gpu.h:174]: We can also advertise Xbox GDK, for One and Series X|S

**Review by flibitijibibo — APPROVED** [review]
A couple minor things for the docs but none of them super important. Part of me wishes there was a way we could sneak in more mentions of console support since I'm sure people will ask, but there's probably a better way to advertise it elsewhere...

File: include/SDL3/SDL_gpu.h (hunk 1 above)
Comment (TheSpydog) [reply include/SDL3/SDL_gpu.h:166]: We can investigate windowless rendering in the future, but currently this is a hard requirement in the Vulkan driver, so even though it's obvious I think it's worth mentioning.

**Review by thatcosmonaut — APPROVED** [review]
+1 on mentioning Xbox and I think we should explicitly mention Vulkan on Nintendo as well. Otherwise looks great

File: include/SDL3/SDL_gpu.h [hunk 3]
```diff
@@ -159,6 +159,24 @@
  * and reasonable to implement across several platforms and underlying APIs.
  * So while these things are not in the "never" category, they are definitely
  * not "near future" items either.
+ *
+ * ## System Requirements
+ *
+ * **Vulkan:** Supported on Windows, Linux, and Nintendo Switch. Requires Vulkan 1.0 with the following extensions and device features:
```
Comment (thatcosmonaut) [inline include/SDL3/SDL_gpu.h:165]: Maybe also mention "certain Android devices" here?

**Comment by TheSpydog** [comment]
Updated to mention console support, and made the OS version requirements more explicit!

File: include/SDL3/SDL_gpu.h (hunk 3 above)
Comment (TheSpydog) [reply include/SDL3/SDL_gpu.h:165]: Oh gosh I completely erased Android from my mind... will fix, thanks

File: include/SDL3/SDL_gpu.h [hunk 4]
```diff
@@ -159,6 +159,24 @@
...
+ * - `VK_KHR_swapchain`
+ * - `VK_KHR_maintenance1`
+ * - `independentBlend`
+ * - `imageCubeArray`
+ * - `depthClamp`
+ * - `shaderClipDistance`
+ * - `drawIndirectFirstInstance`
+ *
+ * **D3D12:** Supported on Windows 10 or newer, Xbox One (GDK), and Xbox Series X (GDK). Requires a GPU that supports DirectX 12 Feature Level 11_1.
```
Comment (flibitijibibo) [inline include/SDL3/SDL_gpu.h:174]: Teeny change but it turns out they're particular about the naming:

```suggestion
 * **D3D12:** Supported on Windows 10 or newer, Xbox One (GDK), and Xbox Series X|S (GDK). Requires a GPU that supports DirectX 12 Feature Level 11_1.
```

File: include/SDL3/SDL_gpu.h (hunk 3 above)
Comment (thatcosmonaut) [reply include/SDL3/SDL_gpu.h]: Believe me, I would like to as well!

**Comment by maia-s** [comment]
I can't find any information about which devices support the Mac2 gpu family. Is it only supported on apple silicon?

**Comment by thatcosmonaut** [comment]
I've tested the GPU API on my 2012 Macbook Pro which has an Nvidia card in it, so maybe that's some indication? Apple is pretty unclear about this.

**Comment by TheSpydog** [comment]
The only place where Apple really defines what a "Mac2" device is is on the [deprecated feature set page here](https://developer.apple.com/documentation/metal/mtlfeatureset/mtlfeatureset_macos_gpufamily2_v1?language=objc). I've added a link to this page in the comment, hopefully that clarifies things. (We'll just have to hope that Apple doesn't take that page down anytime soon...)

**Review by slouken — APPROVED** [review]

**Pull request merged**